### DISCUSSION TITLE
Error: invalid grant

### DISCUSSION THREAD

**Discussion by davidbrochart** [fief-dev/fief#231] [Q&A]
I deployed a public server, where NGINX is proxying my FastAPI application that runs at http://127.0.0.1:8000. I previously tested my application on a local setup and authentication was working fine with Fief. The only thing I changed in Fief is the redirect URI, to point to the public server at https://my.public.domain/auth-callback, and I made the equivalent change here:
```py
@app.get("/auth-callback", name="auth_callback")  
async def auth_callback(request: Request, response: Response, code: str = Query(...)):
    # redirect_uri = request.url_for("auth_callback")
    redirect_uri = "https://my.public.domain/auth-callback"
    ...
```
But now Fief fails after I click "Finalize sign up":
```
fief_client.client.FiefRequestError: [400] - {"error":"invalid_grant"}
```
Do you have any idea what could go wrong? I know it's not a lot of information, but I have no clue where to investigate.

**Comment by davidbrochart** [comment]
Ah my bad, there was another place where I had to replace the redirect URI, sorry for the noise!

**Comment by agademer** [comment]
@davidbrochart : You saved my day :)

To be explicit, the `redirect_uri ` appears at least in `auth_callback ` and in `get_unauthorized_response` functions. You need to make the same change to both if you want your authentication to work, and the `invalid_grant` error message tells you that probably both are not identical :)
